PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/maven/MavenDependencyManager.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) Microsoft Corporation
+ *
+ * All rights reserved.
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.intellij.maven;
+
+import com.microsoft.intellij.util.PomXmlUpdater;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.maven.artifact.versioning.DefaultArtifactVersion;
+import org.dom4j.*;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class MavenDependencyManager {
+    public static final String POM_NAMESPACE = "http://maven.apache.org/POM/4.0.0";
+
+    protected Document doc;
+
+    public MavenDependencyManager(String effectivePomXml) throws DocumentException {
+        Map<String, String> nsContext = new HashMap<>();
+        nsContext.put("ns", POM_NAMESPACE);
+        DocumentFactory.getInstance().setXPathNamespaceURIs(nsContext);
+        doc = DocumentHelper.parseText(effectivePomXml);
+    }
+
+    public String getPluginConfiguration(String groupId, String artifactId, String configurationName) {
+        for (Node node : doc.selectNodes("//ns:project/ns:build/ns:plugins/ns:plugin")) {
+            String myGroupId = ((Element) node).elementTextTrim("groupId");
+            String myArtifactId = ((Element) node).elementTextTrim("artifactId");
+            if (StringUtils.equals(groupId, myGroupId) && StringUtils.equals(artifactId, myArtifactId)) {
+                Element configurationNode = ((Element) node).element("configuration");
+                return configurationNode == null ? null : configurationNode.elementTextTrim(configurationName);
```

Comment:
Is there any constant variables for "groupId", "artifactId" and "configuration" ?
I guess there would be some other classes need to reference them.

Reply:
resolved